components/camera/collectors.go

```diff
@@ -120,6 +124,61 @@ func newReadImageCollector(resource interface{}, params data.CollectorParams) (d
 	return data.NewCollector(cFunc, params)
 }
 
+func newReadImagesCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
+	camera, err := assertCamera(resource)
+	if err != nil {
+		return nil, err
+	}
+	// choose the best/fastest representation
+	mimeType := params.MethodParams["mime_type"]
+	if mimeType == nil {
+		// TODO: Potentially log the actual mime type at collector instantiation or include in response.
+		strWrapper := wrapperspb.String(utils.MimeTypeRawRGBA)
+		mimeType, err = anypb.New(strWrapper)
+		if err != nil {
+			return nil, err
+		}
+	}
+	cFunc := data.CaptureFunc(func(ctx context.Context, _ map[string]*anypb.Any) (interface{}, error) {
+		_, span := trace.StartSpan(ctx, "camera::data::collector::CaptureFunc::ReadImages")
+		defer span.End()
+
+		ctx = context.WithValue(ctx, data.FromDMContextKey{}, true)
+
+		imgs, res, err := camera.Images(ctx)
+		if err != nil {
+			if errors.Is(err, data.ErrNoCaptureToStore) {
+				return nil, err
+			}
+			return nil, data.FailedToReadErr(params.ComponentName, readImages.String(), err)
+		}
+
+		mimeStr := new(wrapperspb.StringValue)
```

Comment:
Since mime type is inferred from the images when they're requested from the camera (see [here](https://github.com/viamrobotics/rdk/blob/415bd01d8daed31437132c86d46f3cd05a57b056/components/camera/server.go#L113C2-L118)), can use the mime type format that you're now grabbing from `encodeImageFromUnderlyingType` below and remove `mime_type` here and from additional params.